### ISSUE TITLE
Iphone 15 compatible via troll store on 17.0 with no adapter (nxboot 0.3.2)

### ISSUE THREAD

**Issue by thingythings** [mologie/nxboot#12]
I can confirm that with no adapter, solely an apple usbc to usbc cable cable of data & power, the phone connects to the switch and can successfully inject payloads (tested with hekate).
![IMG_0396](https://github.com/user-attachments/assets/d5850298-3626-4139-b7c5-b281402828dd)

This works reliably and immediately, tested with an unpatched switch (HAC-001).

**Comment by mologie** [comment]
Thanks so much for the report :)

Happy to hear that this works, I have updated the FAQ on the website accordingly.

**Comment by jacobprezant** [comment]
Great, thanks!

**Comment by jacobprezant** [comment]
This is more of a general point/question, but I believe it would likely work without an adapter on usbc iPads- furthermore, possibly non TrollStore signed with a certificate with usb driver kit entitlements? I see that you’ve mentioned it needing to run on an unsandboxed installer- wasn’t sure if you were aware of this entitlement. (Thought it only supports M series devices- ie not iOS)
Have you considered an iPad version for nxboot?

**Comment by mologie** [comment]
You are right it might work with USB driver kit, but I have not investigated further whether USB driver kit provides the level of access required or is some abstraction instead, nor do I expect Apple to play along much and let me into the mobile App Stores with NXBoot (maybe macOS though). Me alone having the entitlement would not be too useful if nobody else could use it on stock iPads.

**Comment by jacobprezant** [comment]
Thanks for your response
You're right, apple would likely not allow something like this onto the App Store, but I am aware of several GitHub projects with ipas that require additional entitlements, and need to be signed with an apple developer account.
I would assume that because the switch is not class compliant, it would need to use some sort of custom IOUSBHOST driver, integrated with usbdriverkit, subclassed under IOService?

Just some ideas

TLDR; with a custom driver maybe?

**Comment by mologie** [comment]
Surely this needs a custom driver, but still needs entitlements hard to obtain for most. Given all the hassle associated with it I'd rather focus on the macOS side of things when putting time into the project, seems like that will reach a wider audience.

**Comment by jacobprezant** [comment]
Great, thanks again for your input.
